### ISSUE TITLE
nvda should announce the next item  automatically when i right click  of any windows.

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4231]
**Reported by shekhargupta1121 on 2014-06-28 04:32**
1: move to the desktop.
2: right click of any blank area of desktop screen.
you will hear  the only context menu, not next item
want to    announce next item or line automatically  when i right click

**Comment by bhavyashah** [comment]
I am not too sure if it is NVDA interfering, or Windows itself only launches the context menu without focusing on the first menu item. Still, from a UX viewpoint, this sounds desirable, particularly for a newbie or one transitioning from elsewhere, and I vaguely recall being confused myself about why NVDA only said 'context menu' and did not report any menu item a few years ago, although that might just be an illusion. :) All in all, I am in favour of this request, whether viewed as a new feature or a bug fix, to be accepted and triaged. @feerrenrut

**Comment by feerrenrut** [comment]
Visually, no items are highlighted when you first open the context menu. I think this might be risky to do, there are some situations where typing will automatically open a context menu EG auto complete. Sometimes focusing an item will trigger an action EG preview the outcome of the autocomplete. 

We need to better understand the use case here. Why is it confusing when you don't hear the first item in the menu? Where does the expectation of this behaviour come from?

**Comment by josephsl** [comment]
Hi, partly because of the fact that certain screen readers do announce the first item of a context menu when it opens. This small thing makes a huge difference in user experience for some. I’m of the opinion that it is better to follow what’s on screen for context menus (as Reef points out). Thanks.

**Comment by Adriani90** [comment]
This is a duplicate of an issue I have created. In my opinion NVDA should Report the first menu item even though it is not highlighted but it is not a big priority.

**Comment by josephsl** [comment]
Hi,

My word still stands: we want accuracy in regards to menu item selection, not dictate that screen readers may announce the first item regardless of selection state. I'm also interested in seeing counterexamples as noted in the closed ticket.

CC @MichaelDCurran

**Comment by CyrilleB79** [comment]
IMO the first menu item should neither be reported nor selected and we should inform on what Windows does, not change Windows/Office behaviour.

* If it is selected artificially by NVDA whereas Windows would not select it when NVDA is not loaded, there is a risk of side effect as described by @feerrenrut with the auto-complete example
* if it is reported but not selected, the user may think that this item is indeed selected and that pressing Enter key would execute the corresponding action, what is not true. E.g.:

  * example 1: try to right click on the desktop where there is no icon (as explained in initial description) and validate with Enter.
  * example 2 (without the need to do a mouse click): in Windows Explorer, selecte an item, then press Ctrl+space to unselect it, press application key to open menu and press Enter.

In both cases, no action is triggered; the action corresponding to the first menu item is not executed.

I have had a look at #8071 (duplicate of this one that has been closed). Here are my feedback on some points that have not been clarified there.

> It would be good to get sighted confirmation that in Outlook, the
context menu does or doesn't have its first item focused before pressing
an arrow key. 

With Outlook 2016: I confirm that first item is not selected.

> in Jaws for example, I think the first menu entry is spoken when pressing application key. For example on a desktop element “open” on an outlook message “fast print”.

On Outlook 2016, pressing app key, I confirm the first menu item is selected when Jaws is running! This is not the case if NVDA is running or if no screen reader is running. Jaws is probably doing something to have it selected. Again, I do not think that NVDA should modify Windows or Office native behaviour.

> Well in this case NVDA should not announce the first item in the explorer when pressing win+e. Because you have to press an arrow key to focus an item.

Yes it should. When you open Windows Explorer, the first element has focus but is not selected. NVDA reports the name of this element since it has the focus and it also reports that this element is unselected. All seems correct to me.

**Comment by Adriani90** [comment]
Ok it seems there is a consensus regarding this among the developers and also throughout the community. I will close this issue for now. But if you have strong arguments as why NVDA should modify windows behavior, please comment on this issue and we can reopen it.
